backend-rust/src/indexer.rs

```diff
@@ -876,7 +915,186 @@ WHERE blocks.finalization_time IS NULL AND blocks.height <= last.height
     }
 }
 
-/// Prepared block item (transaction), ready to be inserted in the database.
+struct PreparedBlockSpecialEvent {
+    account_address:  String,
+    amount:           i64,
+    block_height:     i64,
+    slot_time:        DateTime<Utc>,
+    transaction_type: AccountStatementEntryType,
+}
+
+impl PreparedBlockSpecialEvent {
+    fn prepare(
+        event: &SpecialTransactionOutcome,
+        slot_time: DateTime<Utc>,
+        block_height: AbsoluteBlockHeight,
+    ) -> anyhow::Result<Vec<Self>> {
+        let results: Box<dyn Iterator<Item = Result<Self, anyhow::Error>>> = match &event {
```

Comment:
This should be rewritten into a Vec. This is the better solution, but would have to be propagated into the database layer to make sense.